internal/rbd/nodeserver.go

```diff
@@ -718,8 +719,24 @@ func (ns *NodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 	}
 
 	diskMounter := &mount.SafeFormatAndMount{Interface: ns.mounter, Exec: utilexec.New()}
-	// TODO check size and return success or error
 	volumePath += "/" + volumeID
+
+	// check size of the mount path
+	var vStat syscall.Statfs_t
+	err = syscall.Statfs(volumePath, &vStat)
```

Comment:
This asks for the size that is available in the filesystem, but you'll need to check the size of the block device instead. Filesystems have some overhead, so the available size on the filesystem is less than was requested for the PVC. Depending on the size of the block device, and type of filesystem used, the different may run into many MBs.

Not sure how you can check that a filesystem occupies (near to) the full block ...

Reply:
I think RoundOffBytes will take care of that. Also, block device would be already resized by this time and always be equal to the requested size.

So, IMO, this is the best we can do.

Reply:
I do not think `RoundOffBytes` will do it sufficiently. You can try to  create a very small PVC (1 MB, and resize it to 2 MB). Or create a very large PVC (multiple TBs and double the size). Both with XFS and ext4... There will be a large difference in the filesystem overhead, and the `vPathSize == volSize` check may never get to `true`.

Reply:
Hmm.. we don't have previous size also here otherwise we could have checked if latest size is greater than that.

Then AFAICT, there is no good way of doing this and we should remove this check.

@humblec any comments?

Reply:
Actually we could check the size and compare between `volume_path and staging_target_path `and decide we already got it covered. Isnt it ? I am fine to remove this check too.   @agarwal-mudit

Reply:
1. Those both would have same size i.e. the previous size
2. We don't have both in kube <v19.0

Reply:
Lets remove this check.. Its fine if we tried resize again.. it will be NO OP.